build/build.go

```diff
@@ -1235,6 +1238,33 @@ func LoadInputs(ctx context.Context, d driver.Driver, inp Inputs, pw progress.Wr
 			target.FrontendAttrs["context:"+k] = v.Path
 			continue
 		}
+
+		// handle OCI layout
+		if strings.HasPrefix(v.Path, "oci-layout://") {
+			pathAlone := strings.TrimPrefix(v.Path, "oci-layout://")
+			// need to add "dummy.io", because reference.Parse*() funcs require we do not start with a '/'
+			named, err := reference.ParseNormalizedNamed(fmt.Sprintf("dummy.io%s", pathAlone))
+			if err != nil {
+				return nil, errors.Wrapf(err, "invalid oci-layout context %s, must be oci-layout:///path/to/layout@sha256:hash", v.Path)
+			}
+			// check that we had a digest
+			digested, ok := named.(reference.Digested)
+			if !ok {
+				return nil, errors.Wrapf(err, "invalid oci-layout context %s, must be oci-layout:///path/to/layout@sha256:hash", v.Path)
+			}
+			// now ensure that we start with a `/`
+			localPath := fmt.Sprintf("/%s", reference.Path(named))
+			store, err := local.NewStore(localPath)
+			if err != nil {
+				return nil, errors.Wrapf(err, "invalid store at %s", localPath)
+			}
+			// now we can add it
+			target.OCIStores = map[string]content.Store{
```

Comment:
This needs to add a store, not replace the whole map

Reply:
Good point. It should check if the map is nil, if so create it, and then add it either way.